internal/pkg/client/request.go

```diff
@@ -157,7 +158,16 @@ func (r *Request) SetContext(ctx context.Context) *Request {
 func (r *Request) WaitFor(subRequest *Request) {
 	r.lock.Lock()
 	defer r.lock.Unlock()
+
+	// Stop the execution of the listeners
+	if r.cancelListeners != nil {
+		r.cancelListeners()
```

Comment:
Riesenim je explicitne zastavit vykonavanie aktualne beziaciej metody `invokeListeners`.

Reply:
Tj. nespoliehat sa na to, ze ju ukonci dalsie volanie `nextListener`, pretoze existuje nedokonceny sub-request.
Problem je v tom, ze kym sa program na to miesto dostane, tak sub-request moze byt uz dokonceny.